health.go

```diff
@@ -1088,9 +1088,9 @@ type TLSCert struct {
 type MinioHealthInfo struct {
 	Error string `json:"error,omitempty"`
 
-	Config      MinioConfig   `json:"config,omitempty"`
-	Info        MinioInfo     `json:"info,omitempty"`
-	Replication *ReplDiagInfo `json:"replication,omitempty"`
+	Config      MinioConfig     `json:"config,omitempty"`
+	Info        MinioInfo       `json:"info,omitempty"`
+	Replication *ReplDiagInfoV2 `json:"replication,omitempty"`
```

Comment:
Is it breaking, if so it should be separate. If not, there is no point in making a `ReplDiagInfoV2` struct.

If breaking... do

```suggestion
	Replication *ReplDiagInfo `json:"replication,omitempty"`
	ReplicationV2 *ReplDiagInfoV2 `json:"replicationV2,omitempty"`
```

If not

```suggestion
	Replication *ReplDiagInfo `json:"replication,omitempty"`
```
.. and add new fields to the existing struct.

Reply:
Actually `eos` side changes not yet merged, but I understand if with merged changes in `madmin-go` released we should add new ones.

Reply:
>  but I understand if with merged changes in madmin-go released we should add new ones.

Sorry. I can't parse what you mean by that.

* If your changes are *incompatible*, make it a new JSON field and allow to un-marshal either.
* If your changes are *compatible*, just add the new stuff to existing. Drop the V2 structs.

It is one of the above. You should chose.

Reply:
Changes are actually incompatible but old `ReplDiagInfo` not yet used in `eos` code. In this case is it like, retain the name as `ReplDiagInfo` only change the attributes of the struct?

Reply:
Yeah, let's just drop the v2 then  and use this as v1👍🏼